$NetBSD: patch-aa,v 1.6 2015/11/07 23:39:15 dholland Exp $

- configure for pkgsrc
- provide a correct all: target
- remove --traditional-cpp on MacOS as it breaks the build
- remove custom snprintf on Solaris as it hasn't been needed in ages
- honor pkgsrc LDFLAGS

--- Makefile.orig	2003-05-30 23:42:14.000000000 +0000
+++ Makefile
@@ -1,5 +1,7 @@
 TARGET = malsync
 
+all: ${TARGET}
+
 PLAT := $(shell uname)
 	
 ifeq ($(PLAT),SunOS)
@@ -15,14 +17,13 @@ LINKLIBS = -lm
 endif
 
 TREETOP = .
-PILOT_LINK_DIR = $(TREETOP)/pilot-link-0.11.7
 
 CFLAGS = -Wall -g -DMALSYNC -I$(TREETOP)/mal/common \
   -I$(TREETOP)/mal/client/common -I$(PILOT_LINK_DIR)/include \
   -I$(TREETOP)/mal/client/unix
 
 ifeq ($(PLAT),Darwin)
-CFLAGS += --traditional-cpp -DDARWIN -D__unix__ -D__FreeBSD__ -D_HPUX_SOURCE
+CFLAGS += -DDARWIN -D__unix__ -D__FreeBSD__ -D_HPUX_SOURCE
 endif
 
 SRC = malsync.c  \
@@ -59,10 +60,6 @@ SRC = malsync.c  \
   $(TREETOP)/mal/client/common/AGSyncProcessor.c \
   $(TREETOP)/mal/common/AGServerConfig.c
 
-ifeq ($(PLAT),SunOS)
-SRC += ap_snprintf.c
-endif
-
 OBJS = $(SRC:.c=.o)
 
 ifeq ($(PLAT),Darwin)
@@ -72,8 +69,8 @@ DYNLINK = -rdynamic
 endif
 
 $(TARGET): $(OBJS)
-	$(CC) $(DYNLINK) $(CFLAGS) -o $@ $(OBJS) \
-          ${PILOT_LINK_DIR}/libpisock/.libs/libpisock.a $(LINKLIBS)
+	$(CC) $(DYNLINK) $(CFLAGS) $(LDFLAGS) -o $@ $(OBJS) \
+          -L${PILOT_LINK_DIR}/lib -lpisock $(LINKLIBS)
 clean:
 	rm -f $(OBJS) $(TARGET)
 
